### ISSUE TITLE
Search shows Backup entries, regardless of Search preference.

### ISSUE THREAD

**Issue by mkohler** [Kunzisoft/KeePassDX#320]
I am seeing a problem in 2.5.0.0beta22 that is not in 2.5.0.0beta18.

### To Reproduce
1. Open KeePass DX.
2. Open and unlock a database with my finger.
3. Click "All".
4. Click search icon.
5. Start typing an entry that has many Backup entries, until you have typed enough letters that the entry is unique.

### What Happens:
1. I see 6 copies of my entry. (Presumably I'm seeing historical entries from the backup folder.)
2. If I click the enter key to complete the search, I see even more entries.

### What Should Happen
1. If I have "Don't search through backup entries" checked, I should just see one entry.

### Additional Notes

1. I have tried changing the setting "Don't search through backup entries" but it appears to have no effect.

**Comment by J-Jamet** [comment]
Thank you @mkohler for the feedback. I think the problem comes from the new indexing system. Can you tell me if you noticed the problem on a database V1 or / and V2?

**Comment by mkohler** [comment]
Based on the command below, V2.
```
$ file home.kdbx 
home.kdbx: Keepass password database 2.x KDBX
$
```

**Comment by J-Jamet** [comment]
I just checked and the current algorithm does not retrieve the items in the RecycleBin folder when the option is checked. There is also no bug with the history for this. Are you sure your backup folder matches the recycle bin? (Where does your entry go when it is deleted?)

**Comment by J-Jamet** [comment]
If there are only 6 entries in the quick search this is normal, it is because it is the maximum number of display. With a V2 database, the complete search displays all entries in the database except those in the RecycleBin.

**Comment by mkohler** [comment]
When I reported the bug, I was not distinguishing between the Backup and RecycleBin folders in my mind. I rarely use either, so I just hadn't thought about the differences. Now, after reviewing how both work, I meant the backup folder.

I'll change the bug description to reflect that.

**Comment by mkohler** [comment]
I upgraded back to 2.5.0.0beta22 and confirmed I can still reproduce the problem. Let me describe the situation more clearly.

### 2.5.0.0beta18 and 2.5.0.0beta22

If I scroll through the All folder, I see 1 entry for "Foo".
If I scroll through the Backup folder, I see 10 entries for "Foo".
If I scroll through the RecycleBin folder, I see 1 entry for "Foo Company".

### 2.5.0.0beta18

If I search for "foo", I see 1 entry for "Foo". (as expected)

### 2.5.0.0.beta22

If I search for "foo", I see 1 entry for "Foo Company" and 10 entries for "Foo".

**Comment by J-Jamet** [comment]
Okay, I see the problem. The "Backup" folder is normally used only for V1 databases. V2 databases no longer use this system but a history for each entry and a recycle bin. Your database should have been previously a V1 you imported and the new algorithm only filters the backup folder for V1 databases. 
I checked the behavior on KeePass (windows) for this kind of case and it keeps the filter of the folders "backup" for the databases V2. The old code of KeePassDX2.5.0.0beta18 (from KeePassDroid) was very messy, so I had separated things well and unfortunately remove this compatibility. I'm going to add the condition for the V2 databases for the next update.
